yarn-project/pxe/src/note_processor/produce_note_dao.ts

```diff
@@ -89,10 +98,72 @@ export async function produceNoteDaos(
           txHash,
           noteHashes,
           dataStartIndexForTx,
+          unencryptedLogs,
         );
       }
+    } else if (
+      (e as any).message.includes('failed to solve blackbox function: embedded_curve_add') ||
+      (e as any).message.includes('Could not find key prefix.')
+    ) {
+      // TODO(#8769): This branch is a temporary partial notes delivery solution that should be eventually replaced.
```

Comment:
Yes, it was terrible. I did the following to the code:

1. Improved code reuse by implementing `produceNoteDaoForKey` so that we don't have doubled logic for outgoing and incoming,
2. separated the functions into multiple files,
3. separated the partial note logic into a function.

The code is still not exactly beautiful but hopefully will not make you want to spoon out your eyes.